tests/reporting/reporting_utils.py

```diff
@@ -0,0 +1,168 @@
+from ethereum.tools import tester
+from ethereum.tools.tester import TransactionFailed
+from pytest import fixture, mark, lazy_fixture, raises
+from datetime import timedelta
+
+def initializeReportingFixture(sessionFixture, market):
+    # Seed legacy rep contract
+    legacyRepContract = sessionFixture.contracts['LegacyRepContract']
+    legacyRepContract.faucet(long(11 * 10**6 * 10**18))
+    branch = sessionFixture.branch
+
+    # Get the reputation token for this branch and migrate legacy REP to it
+    reputationToken = sessionFixture.applySignature('ReputationToken', branch.getReputationToken())
+    legacyRepContract.approve(reputationToken.address, 11 * 10**6 * 10**18)
+    reputationToken.migrateFromLegacyRepContract()
+
+    # Give some REP to testers to make things interesting
+    for testAccount in [tester.a1, tester.a2, tester.a3]:
+        reputationToken.transfer(testAccount, 1 * 10**6 * 10**18)
+
+    reportingWindow = sessionFixture.applySignature('ReportingWindow', market.getReportingWindow())
+    firstRegistrationToken = sessionFixture.applySignature('RegistrationToken', reportingWindow.getRegistrationToken())
+    reportingTokenNo = sessionFixture.getReportingToken(market, [2,0])
+    reportingTokenYes = sessionFixture.getReportingToken(market, [0,2])
+
+    # Tester 0, 1, 2, and 3 will buy registration tokens so they can report on the market
+    firstRegistrationToken.register(sender=tester.k0)
+    assert firstRegistrationToken.balanceOf(tester.a0) == 1
+    # Testers will have their previous REP balance less the registration token bond of 1 REP
+    assert reputationToken.balanceOf(tester.a0) == 8 * 10**6 * 10**18 - 10**18
+
+    for (key, address) in [(tester.k1, tester.a1), (tester.k2, tester.a2), (tester.k3, tester.a3)]:
+        firstRegistrationToken.register(sender=key)
+        assert firstRegistrationToken.balanceOf(address) == 1
+        assert reputationToken.balanceOf(address) == 1 * 10**6 * 10**18 - 10**18
+
+    return sessionFixture.chain.snapshot()
+
+def proceedToAutomatedReporting(testFixture, market):
+    cash = testFixture.cash
+    branch = testFixture.branch
+    reputationToken = testFixture.applySignature('ReputationToken', branch.getReputationToken())
+    reportingWindow = testFixture.applySignature('ReportingWindow', market.getReportingWindow())
+
+    # We can't yet do an automated report on the market as it's in the pre reporting phase
+    if (market.getReportingState() == testFixture.constants.PRE_REPORTING()):
+        with raises(TransactionFailed, message="Reporting cannot be done in the PRE REPORTING state"):
+            market.automatedReport([0,2], sender=tester.k0)
```

Comment:
This will only work with binary market.  Consider taking in the array as a parameter to this function so this will work with n-outcome markets and scalar markets. (same comment elsewhere in this file).